model/src/main/kotlin/utils/PurlExtensions.kt

```diff
@@ -71,7 +71,14 @@ fun Identifier.getPurlType() =
  */
 @JvmOverloads
 fun Identifier.toPurl(qualifiers: Map<String, String> = emptyMap(), subpath: String = "") =
-    if (this == Identifier.EMPTY) "" else createPurl(getPurlType(), namespace, name, version, qualifiers, subpath)
+    if (this == Identifier.EMPTY) {
+        ""
+    } else {
+        val combined = "$namespace/$name"
+        val purlNamespace = combined.substringBeforeLast('/')
+        val purlName = combined.substringAfterLast('/')
```

Comment:
Hm, I wonder for e.g. the Go packages, with this `v2` or `vn` at the end, if this is correct. See e.g.
https://stackoverflow.com/questions/53344471/taking-a-repository-to-v2. Would that mean the package name is `v2` and the rest is the namespace?

Reply:
Good question, I have no idea. Maybe it depends on the approach (branches vs. sub-directories) chosen for major version bumps?

The only purl for a Go package with a major version > 1 I could find in our test data is [this](https://github.com/oss-review-toolkit/ort/pull/9330/files#diff-5871a47ebd961d0c891d76056ddeb3aeff1193ff70e0f229802543237bc36465R486), which uses `yaml.v3` instead of `yaml/v3` for the name. So maybe already Go itself replaces `/vN` with `.vN` when getting data for dependencies?